jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -280,6 +370,7 @@ public InvocationType getInvocationType()
                 }
             };
 
+            // If the fields are already available, we can process from this handling thread
             FormFields.onFields(getRequest(), _charset, onFields);
```

Comment:
I know this is not from this PR but seems weird that we have a bunch of APIs on `FormFields` taking `Promise.Invocable<Fields>` and a bunch returning `CompletableFuture<Fields>`. 

Seems like we should just pick one.

Reply:
We have.  The CF APIs are all deprecated.